### PR TITLE
Tests/kch fix failing on windows and stabilize

### PR DESCRIPTION
**Pull request by kamilchodola**
**Fixes** | Closes | Resolves 
No Issue number - adjusting tests on Windows and ubuntu - part1

## Changes:
- Added missing tests (5 projects). @flcl42 and @LukaszRozmej adjusted Db and KeyStore tests to pass.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [Y] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by rubo** [comment]
I'm not sure why this PR includes other not-yet merged PRs. If it should include, then we need to close those PRs.

File: .github/workflows/run-nethermind-tests.yml [hunk 1]
```diff
@@ -201,6 +201,42 @@ jobs:
       run: |
         dotnet test -c Release src/Nethermind/Nethermind.State.Test
 
+  neth-tests6:
+    name: Running Nethermind Tests 6
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (rubo) [inline .github/workflows/run-nethermind-tests.yml:208]: Please update to `actions/checkout@v3`.

File: .github/workflows/run-nethermind-tests.yml [hunk 2]
```diff
@@ -201,6 +201,42 @@ jobs:
...
+      with:
+        submodules: true
+    - name: Setting up dotnet
+      uses: actions/setup-dotnet@v1
```
Comment (rubo) [inline .github/workflows/run-nethermind-tests.yml:212]: Please update to `actions/setup-dotnet@v2`.

**Review by rubo — CHANGES_REQUESTED** [review]

**Comment by kamilchodola** [comment]
@rubo @flcl42 @LukaszRozmej @MarekM25 @marcindsobczak 
Would ask for re-review.

Removed other changes from Alexey and Łukasz since those are already merged to master on other PRs and made small improvements as suggested by Ruben.
Additionally, I've added those 5 missing projects to codecoverage pipeline.

File: .github/workflows/run-nethermind-tests.yml (hunk 2 above)
Comment (kamilchodola) [reply .github/workflows/run-nethermind-tests.yml]: Done :) Thx!

File: .github/workflows/run-nethermind-tests.yml (hunk 1 above)
Comment (kamilchodola) [reply .github/workflows/run-nethermind-tests.yml]: Done :) Thx!

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**